extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/validator/LDAPCredentialsValidator.java

```diff
@@ -0,0 +1,420 @@
+        credentialMaxDuration == null ? BasicAuthUtils.DEFAULT_CREDENTIAL_MAX_DURATION_SECONDS : credentialMaxDuration,
+        credentialCacheSize == null ? BasicAuthUtils.DEFAULT_CREDENTIAL_CACHE_SIZE : credentialCacheSize
+    ));
+    this.cache = new LruBlockCache(
+        this.dbConfig.get().getCredentialCacheSize(),
+        this.dbConfig.get().getCredentialVerifyDuration(),
+        this.dbConfig.get().getCredentialMaxDuration()
+    );
+  }
+
+  Properties bindProperties(BasicAuthDBConfig dbConfig)
+  {
+    Properties properties = commonProperties(dbConfig);
+    properties.put(Context.SECURITY_PRINCIPAL, dbConfig.getBindUser());
+    properties.put(Context.SECURITY_CREDENTIALS, dbConfig.getBindPassword().getPassword());
+    return properties;
+  }
+
+  Properties userProperties(BasicAuthDBConfig dbConfig, LdapName userDn, char[] password)
+  {
+    Properties properties = commonProperties(dbConfig);
+    properties.put(Context.SECURITY_PRINCIPAL, userDn.toString());
+    properties.put(Context.SECURITY_CREDENTIALS, String.valueOf(password));
+    return properties;
+  }
+
+  Properties commonProperties(BasicAuthDBConfig dbConfig)
+  {
+    Properties properties = new Properties();
+    properties.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+    properties.put(Context.PROVIDER_URL, dbConfig.getUrl());
+    properties.put(Context.SECURITY_AUTHENTICATION, "simple");
+    if (StringUtils.toLowerCase(dbConfig.getUrl()).startsWith("ldaps://")) {
+      properties.put(Context.SECURITY_PROTOCOL, "ssl");
+      properties.put("java.naming.ldap.factory.socket", BasicSecuritySSLSocketFactory.class.getName());
+    }
+    return properties;
+  }
+
+  @Override
+  public AuthenticationResult validateCredentials(
+      String authenticatorName,
+      String authorizerName,
+      String username,
+      char[] password
+  )
+  {
+    updateConfig(authenticatorName);
+    BasicAuthDBConfig currentDBConfig = this.dbConfig.get();
+    Set<LdapName> groups;
+    LdapName userDn;
+    Map<String, Object> contexMap = new HashMap<>();
+
+    BasicAuthenticatorUserPrincipal principal = this.cache.getOrExpire(username);
+    if (principal != null && principal.hasSameCredentials(password)) {
+      contexMap.put(BasicAuthUtils.GROUPS_CONTEXT_KEY, principal.getGroups());
+      return new AuthenticationResult(username, authorizerName, authenticatorName, contexMap);
+    } else {
+      try {
+        InitialDirContext dirContext = new InitialDirContext(bindProperties(currentDBConfig));
+
+        try {
+          SearchResult userResult = getLdapUserObject(currentDBConfig, dirContext, username);
+          if (userResult == null) {
+            LOG.debug("User not found: %s", username);
+            return null;
+          }
+          userDn = new LdapName(userResult.getNameInNamespace());
+          groups = getGroupsFromLdap(currentDBConfig, userResult);
+          if (groups == null || groups.isEmpty()) {
```

Comment:
I don't think not having a group assigned should be authentication failure (Suppose a user only cares about using LDAP for authentication and doesn't really care about roles/groups), if the groups are necessary for LDAP-based authorization checks then the missing groups should be checked in the authorizer, not the authenticator

Reply:
Yes I agree too, I'll remove this check/condition. Thanks

Reply:
Fixed, removed check/condition for authentication failure when not having a group assigned